addons/pos_self_order/models/pos_config.py

```diff
@@ -295,7 +295,7 @@ def _get_available_categories(self):
                     *(
                         self.limit_categories
                         and self.iface_available_categ_ids
-                        and [("id", "in", self.iface_available_categ_ids.ids)]
+                        and [("id", "in", self._get_available_categories_ids())]
                         or []
                     ),
                 ],
```

Comment:
can you not remove this function completely?
Then you won't have the problem with the name collision and you can have the function be called `_get_available_categories` and return recordsets, not ids